plugin.video.dailymotion_com/default.py

```diff
@@ -197,45 +217,45 @@ def listVideos(url):
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listVideos', "")
-    if forceViewModeNew:
-        xbmc.executebuiltin('Container.SetViewMode('+viewModeNew+')')
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listVideos', _ipath + "next_page2.png")
+    xbmcplugin.setContent(pluginhandle, "episodes")
+    xbmc.executebuiltin('Container.SetViewMode(54)')
     xbmcplugin.endOfDirectory(pluginhandle)
 
 def listUsers(url):
     content = getUrl2(url)
     content = json.loads(content)
     for item in content['list']:
         if item['username']:
-            user = item['username'].encode('utf-8')
+            user = item['username'].encode('utf-8') if PY2 else item['username']
             thumb = item['avatar_large_url']
             videos = item['videos_total']
             views = item['views_total']
             addUserDir(user, 'owner:'+user, 'sortVideos1', thumb.replace("\\", ""), "Views: "+str(views)+"\nVideos: "+str(videos))
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listUsers', "")
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listUsers', _ipath + "next_page.png")
     xbmcplugin.setContent(pluginhandle, "addons")
+    xbmc.executebuiltin('Container.SetViewMode(500)')
     xbmcplugin.endOfDirectory(pluginhandle)
 
 def listLive(url):
     content = getUrl2(url)
     content = json.loads(content)
     for item in content['list']:
-        title = item['title'].encode('utf-8')
+        title = item['title'].encode('utf-8') if PY2 else item['title']
         vid = item['id']
         thumb = item['thumbnail_large_url']
         views = item['views_last_hour']
         addLiveLink(title, vid, 'playLiveVideo', thumb.replace("\\", ""), views)
     if content['has_more']:
         currentPage = content['page']
         nextPage = currentPage+1
-        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listLive', "")
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listLive', _ipath + "next_page2.png")
     xbmcplugin.setContent(pluginhandle, "episodes")
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```

Comment:
This isn't allowed.

Reply:
Corrected it to use only if user setup the relevant modes in settings